components/blocks/CompareBox/CompareBox.template.ts

```diff
@@ -1,79 +1,148 @@
-import type { Template } from 'tinacms';
+import { useEffect, useState } from "react";
+import { wrapFieldsWithMeta, type Template } from "tinacms";
+import { checkboxList } from "../../../tina/customTinaFormFields/checkboxList";
+
+export const splitOneAndJoin = (item, separator) => {
+    const[first, ...rest] = item.split(separator)
+    return [first, rest.join(separator)]
+}
+
+export const criteriaMapping = (values) => {
+  values.blocks.forEach((block, blockIndex) => {
+    if (block._template === 'CompareBox') {
+    const criteriaItems = block.criteriaItems?.map((item) => item.criteria);
+    block.companies?.forEach((company, index) => {
+      {
+        // Add new criteria to the list of satisfied criteria
+        const oldCriteria = company.satisfiedCriteria?.map((item) => splitOneAndJoin(item, "-")[1]) ?? [];
+        const updatedCriteriaSatisfaction = [];
+        criteriaItems.forEach((item) => {
+          if (oldCriteria.includes(item)) {
+            const satisfaction = company.satisfiedCriteria.find((criteria) => splitOneAndJoin(criteria, "-")[1] === item);
+            updatedCriteriaSatisfaction.push(satisfaction);
+          } else {
+            updatedCriteriaSatisfaction.push(`false-${item}`);
+          }
+        });
+        values.blocks[blockIndex].companies[index].satisfiedCriteria = updatedCriteriaSatisfaction;
```

Comment:
// add comment 're-write original data before save' or something cos this is a scary line